apiserver/facades/client/cloud/cloud.go

```diff
@@ -720,6 +760,33 @@ func (api *CloudAPI) RevokeCredentialsCheckModels(args params.RevokeCredentialAr
 			results.Results[i].Error = common.ServerError(common.ErrPerm)
 			continue
 		}
+
+		models, err := api.credentialModels(tag)
+		if err != nil {
+			if !arg.Force {
+				// Could not determine if credential has models - do not continue updating this credential...
+				results.Results[i].Error = common.ServerError(err)
+				continue
+			}
+			logger.Warningf("could not get models that use credential %v: %v", tag, err)
+		}
+		if len(models) != 0 {
+			// map keys are notoriously randomly ordered
+			uuids := []string{}
+			for uuid := range models {
+				uuids = append(uuids, uuid)
+			}
+			sort.Strings(uuids)
+			for _, uuid := range uuids {
```

Comment:
This could result in a lot of individual warning messages. It would be better to compose a single multi-line message with the model uuids listed and log a single message, eg
```
WARNING: credential foo cannot be deleted as it is used by these models:
 - uuid
 - uuid
```